index.bs

```diff
@@ -2137,6 +2142,54 @@ document loading at which a navigation command will return.
 
 ### Commands ### {#module-browsingContext-commands}
 
+#### The browsingContext.activate Command ####  {#command-browsingContext-activate}
+
+The <dfn export for=commands>browsingContext.activate</dfn> command activates and focuses the given top-level browsing context.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      browsingContext.Activate = {
+        method: "browsingContext.activate",
+        params: browsingContext.ActivateParameters
+      };
+
+      browsingContext.ActivateParameters = {
+        context: browsingContext.BrowsingContext
+      }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl">
+     EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browsingContext.activate">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |context id| be the value of the |command parameters|["<code>context</code>"] field.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=] with |context id|.
+
+1. If |context| is not a [=top-level browsing context=], return [=error=] with [=error code=] [=invalid argument=].
+
+1. Run implementation-specific steps so that |context|'s [=traversable navigable=]'s [=system visibility state=] becomes [=visible=]. If this is not possible return [=error=] with error code [=unsupported operation=].
+
+    Note: This can have the side effect of making currently [=visible=] [=navigables=] [=hidden=].
+
+    Issue: It's not clear if we should have explicit requirements for cases like a minimized window; should that cause the window to become unminimized?
+
+1. Run implementation-specific steps to set the [=top-level traversable/system focus=] on the |context| if it is not focused.
```

Comment:
So, to be clear, this doesn't change the [focused area of the document](https://html.spec.whatwg.org/#focused-area-of-the-document) in the browsing context? That might also be worth calling out.

Reply:
I have been reading spec parts about the focused area of the document but the term is still not completely clear to me. My current understanding is that any DOM element can be the focused area of the document and so I would expect the focused area to change if the system focus + making the  change causes it to: for example, if you insert an element with an autofocus attribute into a background tab (the focused element does not change), as soon as you activate the context, the autofocus applies (it looks like the system focus makes the document fully active and that will apply various focus algorithms based on that). Any suggestions on the best way to describe this here?

Reply:
Added a note that the focused area of the document should not be explicitly changed.

Reply:
OK, I think that's enough. I was thinking of the case where there's already an element with focus, to make it clear that we shouldn't end up with focus on the document (or somewhere in the browser UI).

Reply:
I think this is generally the behaviour in our implementation (no explicit change of the focused area) but we found one edge case is when a new document with an element with the autofocus attribute is inserted while the page is in the background. In that situation, the element gets focused on activation but I believe this is correct according to the HTML spec.

Reply:
If someone inserts an element into a background tab with autofocus, then, although activation will trigger the focus, the initial cause was the person adding the autofocus element into the background tab. They presumably did this knowing this will cause activation to trigger focus, so in truth, activation by itself didn't change the focused area of the document, but a combination of actions. So it still applies that activation (by itself) doesn't trigger a change in focus area.